pkg/asset/manifests/tectonic.go

```diff
@@ -39,8 +41,45 @@ func (t *Tectonic) Generate(dependencies asset.Parents) error {
 	ingressCertKey := &tls.IngressCertKey{}
 	kubeCA := &tls.KubeCA{}
 	dependencies.Get(installConfig, ingressCertKey, kubeCA)
+	var awsCreds AWS
+	var libvirtCreds Libvirt
+	var openStackCreds OpenStack
+	var cloudCredsData []byte
+	switch {
+	case installConfig.Config.Platform.AWS != nil:
+		p := credentials.SharedCredentialsProvider{}
+		awscreds, err := p.Retrieve()
+		if err != nil {
+			return err
+		}
+		awsCreds = AWS{
+			Base64encodeAccessKeyID:     base64.StdEncoding.EncodeToString([]byte(awscreds.AccessKeyID)),
+			Base64encodeSecretAccessKey: base64.StdEncoding.EncodeToString([]byte(awscreds.SecretAccessKey)),
+		}
+	case installConfig.Config.Platform.Libvirt != nil:
+		libvirtCreds = Libvirt{
```

Comment:
These will eventually be TLS credentials, i.e. a certificate and private key. The tricky part is that we probably can't fully manage the CA within the installer because you'll have to configure libvirtd with the CA as part of the initial manual setup.

So, at some point you'll have to either pass in a CA and key and generate the credentials here or just read a previously generated client certificate and key from disk and insert them.

PR #296 was the first pass at TLS for libvirt.